assets/js/posts_report.js

```diff
@@ -0,0 +1,173 @@
+      }
+      if ('' !== to) {
+        filters.before = to + 'T00:00:00';
+      }
+
+      postsView.refreshPosts(filters);
+      pagesView.refreshPages(filters);
+    });
+  });
+
+
+  p4.PostsView = wp.Backbone.View.extend({
+
+    template: wp.template('p4-post-list'),
+
+    events: {
+      'click .refresh': function () {
+        return this.refreshPosts({});
+      },
+    },
+
+    showSpinner: function () {
+      $('#posts_loader').removeClass('hidden');
+    },
+
+    hideSpinner: function () {
+      $('#posts_loader').addClass('hidden');
+    },
+
+    refreshPosts: function (filters) {
+      this.showSpinner();
+      var params = {
+        per_page: 50,
+        status: 'publish',
+        orderby: 'modified',
+        order: 'desc',
+        date_query_column: 'post_modified',
+      };
+      Object.assign(params, filters);
+      this.collection.reset();
+      this.views.remove();
+      this.render();
+      this.collection.fetch({
+        url: p4_data.api_url + '/posts',
+        data: params,
+        headers: {'X-WP-Nonce': p4_data.nonce}
+      });
+      this.hideSpinner();
+    },
+
+    initialize: function () {
+      this.listenTo(this.collection, 'add', this.addPostView);
+    },
+
+    addPostView: function (post) {
+      this.views.add('.p4-posts', new p4.PostView({model: post}));
+    }
+  });
+
+  p4.PostView = wp.Backbone.View.extend({
+    template: wp.template('p4-post'),
+    tagName: 'tr',
+
+    prepare: function () {
+      return this.model.toJSON();
+    }
+  });
+
+
+  p4.PagesView = wp.Backbone.View.extend({
+    template: wp.template('p4-page-list'),
+
+    events: {
+      'click .refresh': function () {
+        return this.refreshPages({});
+      },
+    },
+
+    showSpinner: function () {
+      $('#pages_loader').removeClass('hidden');
+    },
+
+    hideSpinner: function () {
+      $('#pages_loader').addClass('hidden');
+    },
+
+    refreshPages: function (filters) {
+      this.showSpinner();
+      var params = {
+        per_page: 50,
+        status: 'publish',
+        orderby: 'modified',
+        order: 'desc',
+        date_query_column: 'post_modified',
+      };
+      Object.assign(params, filters);
+      this.collection.reset();
+      this.views.remove();
+      this.render();
+      this.collection.fetch({
+        url: p4_data.api_url + '/pages',
+        data: params,
+        headers: {'X-WP-Nonce': p4_data.nonce}
+      });
+      this.hideSpinner();
+    },
+
+    initialize: function () {
+      this.listenTo(this.collection, 'add', this.addPostView);
+    },
+
+    addPostView: function (post) {
+      this.views.add('.p4-pages', new p4.PageView({model: post}));
+    }
+  });
+
+  p4.PageView = wp.Backbone.View.extend({
+    template: wp.template('p4-post'),
+    tagName: 'tr',
+
+    prepare: function () {
+      return this.model.toJSON();
+    }
+  });
+
+  p4.initialize = function () {
+    postCollection = new wp.api.collections.Posts();
```

Comment:
I get `Uncaught TypeError: wp.api.collections.Posts is not a constructor` here, which prevents results from loading.
Am I missing something?

Reply:
I can't reproduce it. Is wp-api.js being loaded?

Reply:
fixed by calling the posts_report.tpl.php call